### PR TITLE
Config backup-count validation

### PR DESCRIPTION
**Pull request by pveentjer**
Adds backup/async-backup validation on configs where missing + removes copy/paste code on places where implemented.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/util/Preconditions.java [hunk 1]
```diff
@@ -148,12 +150,71 @@ public static int checkPositive(int value, String errorMessage) {
...
+        if (newBackupCount < 0) {
+            throw new IllegalArgumentException("backup-count can't be smaller than 0");
+        }
+
+        if (currentAsyncBackupCount < 0) {
+            throw new IllegalArgumentException("async-backup-count can't be smaller than 0");
+        }
+
+        if (newBackupCount > MAX_BACKUP_COUNT) {
+            throw new IllegalArgumentException("backup-count can't be larger than than " + MAX_BACKUP_COUNT);
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/util/Preconditions.java]: is this necessary? the subsequent check will take care of it anyway. I can imagine it produces better error message, but I'm not sure if that's worth having an extra check.

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/util/Preconditions.java]: You detect any potential mess up. So you don't rely on the other method being called..

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/util/Preconditions.java]: what about including passed values in a error message?

**Comment by pveentjer** [comment]
verify

File: hazelcast/src/main/java/com/hazelcast/util/Preconditions.java (hunk 1 above)
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/util/Preconditions.java]: Good idea. Currently that isn't done in the existing code.

I would like to do it in one go for the whole preconditions library

**Pull request merged**